src/sbvr-api/odata-response.ts

```diff
@@ -169,6 +192,20 @@ export const process = async (
 		}
 	}
 
+	if (requiredSigningFields.length > 0) {
+		await Promise.all(
+			rows.map(async (row) => {
+				requiredSigningFields.map(async (fieldName) => {
+					if (configuredWebResourceHandler != null && row[fieldName] != null) {
+						row[fieldName] = await configuredWebResourceHandler.onPreRespond(
+							row[fieldName],
+						);
+					}
+				});
+			}),
+		);
+	}
+
```

Comment:
This should be in line with the loop before, as it otherwise iterates again through all rows. 
Also it should only run if `configuredWebResourceHandler != null` as this is static to the loop.

```suggestion
	if (requiredSigningFields.length > 0) {
		await Promise.all(
			rows.map(async (row) => {
				requiredSigningFields.map(async (fieldName) => {
					if (configuredWebResourceHandler != null && row[fieldName] != null) {
						row[fieldName] = await configuredWebResourceHandler.onPreRespond(
							row[fieldName],
						);
					}
				});
			}),
		);
	}

```

Reply:
"This should be in line with the loop before, as it otherwise iterates again through all rows." I think this is exactly the behavior we want. 
"Also it should only run if configuredWebResourceHandler != null as this is static to the loop." Done!